ToDoApp/ToDoApp/Respository/Respository.swift

```diff
@@ -0,0 +1,43 @@
+//
+//  Respository.swift
+//  ToDoApp
+//
+//  Created by Trong Hieu Vuong on 2022-12-22.
+//
+
+import Foundation
+import FirebaseDatabase
+import FirebaseDatabaseSwift
+protocol Repository {
+    func createOrUpdate<T>(_ value: T) async -> RepositoryError? where T: Identifiable, T: Encodable, T.ID == UUID
+    func read<T>() async -> Result<[T], RepositoryError> where T: Decodable
+}
+enum RepositoryError: Error {
+    case createOrUpdateError
+    case failedToFetchData
+}
+extension DatabaseReference: Repository {
+  // MARK: create a task
+    func createOrUpdate<T>(_ value: T) async -> RepositoryError? where T: Identifiable, T: Encodable, T.ID == UUID {
+        guard let encodedValue = value.toDictionary else {
+            return RepositoryError.createOrUpdateError
+        }
+        do {
+           try await  self.child(value.id.uuidString).setValue(encodedValue)
+        } catch (_) {
+            return RepositoryError.createOrUpdateError
+        }
+        return nil
+    }
+    func read<T>() async -> Result<[T], RepositoryError> where T : Decodable {
+        var task: [T] = []
```

Comment:
Let's avoid this var since it is redundant, but itroduces potential possibilities to mutate it in unexpected way

Reply:
I am not getting this, do you have any suggestion?